presto-elasticsearch/src/main/java/com/facebook/presto/elasticsearch/ElasticsearchTableDescriptionSupplier.java

```diff
@@ -0,0 +1,114 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.elasticsearch;
+
+import com.facebook.presto.spi.PrestoException;
+import com.facebook.presto.spi.SchemaTableName;
+import com.google.common.base.Splitter;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableMap;
+import com.google.common.collect.ImmutableSet;
+import io.airlift.json.JsonCodec;
+
+import javax.inject.Inject;
+
+import java.io.File;
+import java.io.IOException;
+import java.io.UncheckedIOException;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.function.Supplier;
+
+import static com.facebook.presto.elasticsearch.ElasticsearchErrorCode.ELASTIC_SEARCH_TABLE_DEFINITION_ERROR;
+import static com.google.common.base.MoreObjects.firstNonNull;
+import static com.google.common.base.Preconditions.checkArgument;
+import static com.google.common.base.Strings.isNullOrEmpty;
+import static java.nio.file.Files.readAllBytes;
+import static java.util.Objects.requireNonNull;
+
+public class ElasticsearchTableDescriptionSupplier
+        implements Supplier<Map<SchemaTableName, ElasticsearchTableDescription>>
+{
+    private final ElasticsearchConnectorConfig config;
+    private final JsonCodec<ElasticsearchTableDescription> codec;
+
+    @Inject
+    ElasticsearchTableDescriptionSupplier(ElasticsearchConnectorConfig config, JsonCodec<ElasticsearchTableDescription> codec)
+    {
+        this.config = requireNonNull(config, "config is null");
+        this.codec = requireNonNull(codec, "codec is null");
+    }
+
+    @Override
+    public Map<SchemaTableName, ElasticsearchTableDescription> get()
+    {
+        return createElasticsearchTableDescriptions(config.getTableDescriptionDir(), config.getDefaultSchema(), ImmutableSet.copyOf(config.getTableNames()), codec);
+    }
+
+    public static Map<SchemaTableName, ElasticsearchTableDescription> createElasticsearchTableDescriptions(
+            File tableDescriptionDir,
+            String defaultSchema,
+            Set<String> tableNames,
+            JsonCodec<ElasticsearchTableDescription> tableDescriptionCodec)
+    {
+        ImmutableMap.Builder<SchemaTableName, ElasticsearchTableDescription> builder = ImmutableMap.builder();
```

Comment:
Since `tableDefinitions` is a local variable not visible to outside of this method, you can just use a regular `HashMap` and simplify the code a bit.